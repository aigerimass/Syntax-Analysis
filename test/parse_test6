Func_with_While(times){
    a = 0;
    rat = 2;
    while(a < times){
        rat = rat + Read();
        Write(rat);
        a = a + 1;
    };
}

Main () {
    Func_with_While(Read(););
}
