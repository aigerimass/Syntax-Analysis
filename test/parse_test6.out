----------SYNTAX-ANALYSIS----------
Function Definition: [
name = "Func_with_While",
args: times
body:
>Bind(Var "a", 0);

>Bind(Var "rat", 2);

>While(
cond: 
Lt(Var "a", Var "times")
body:
>Bind(Var "rat", Plus(Var "rat", >Call(name: "Read", args: )));
>Call(name: "Write", args: Var "rat");
>Bind(Var "a", Plus(Var "a", 1));
);

]
--------
Function Definition: [
name = "Main",
args: 
body:
>Call(name: "Func_with_While", args: >Call(name: "Read", args: ));

]
--------
----------Run----------------------
14
26
38
50
62
74
86
98
110
122
134
146
----------INTERVAL-ANALYSIS--------
Func_with_While (times) {
#times = [12, 12]
a = 0;
#times = [12, 12];
#a = [0, 0];
rat = 2;
#times = [12, 12];
#a = [0, 0];
#rat = [2, 2];
while (a < times) {
rat = rat + Read ();
#times = [12, 12];
#a = [0, 12];
#rat = [2, 146];
Write (rat);
#times = [12, 12];
#a = [0, 12];
#rat = [2, 146];
a = a + 1;
#times = [12, 12];
#a = [0, 12];
#rat = [2, 146];
};
#times = [12, 12];
#a = [0, 12];
#rat = [2, 146];
};
------------
Main () {
Func_with_While (Read ());
};
------------

